python/pypto/language/parser/ast_parser.py

```
@@ -1186,13 +1189,13 @@ def _parse_submit_assignment(self, target: ast.Tuple, call: ast.Call) -> None:
         """
         span = self.span_tracker.get_span(call)
-        if self._manual_scope_depth == 0:
-            raise ParserSyntaxError(
-                "pl.submit(...) is only valid inside a `with pl.manual_scope():` block",
-                span=span,
-                hint="Wrap the submit in pl.manual_scope(), or use a plain "
-                "self.kernel(...) call outside manual scopes.",
-            )
+        # ``pl.submit`` and ``deps=`` are orthogonal to ``manual_scope``: the
+        # runtime's ``Arg::set_dependencies`` adds explicit edges on top of the
+        # auto-tracked deps (final fanin = auto ∪ explicit), so both flavours
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `∪` (UNION). Did you mean `U` (LATIN CAPITAL LETTER U)?

(RUF003)